airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftSqlOperations.java

```diff
@@ -73,24 +75,14 @@ public void insertRecords(JdbcDatabase database, Stream<AirbyteRecordMessage> re
     SqlOperationsUtils.insertRawRecordsInSingleQuery(insertQueryComponent, recordQueryComponent, database, records);
   }
 
-  public void copyS3CsvFileIntoTable(JdbcDatabase database,
-                                     String s3FileLocation,
-                                     String schema,
-                                     String tableName,
-                                     String s3KeyId,
-                                     String s3Key,
-                                     String s3Region)
-      throws SQLException {
-    final var copyQuery = String.format(
-        "COPY %s.%s FROM '%s'\n"
-            + "CREDENTIALS 'aws_access_key_id=%s;aws_secret_access_key=%s'\n"
-            + "CSV REGION '%s' TIMEFORMAT 'auto';\n",
-        schema, tableName,
-        s3FileLocation,
-        s3KeyId, s3Key,
-        s3Region);
-
-    database.execute(copyQuery);
+  @Override
+  public boolean isValidData(final AirbyteStreamNameNamespacePair streamName, final String data) {
+    final int dataSize = data.getBytes().length;
+    if (dataSize > REDSHIFT_VARCHAR_MAX_BYTE_SIZE) {
+      LOGGER.warn("Data({}) from stream {} exceeds limit of VARCHAR(65535) on Redshift... Ignoring record", dataSize, streamName);
```

Comment:
I imagine usually if someone is running into this problem it's probably pervasive across that table. If it's a large table we don't want to log once per record.

I'm fine merging as is as long as we have a ticket to fix this problem.